### DISCUSSION TITLE
[Feat Req] Time units should support plurals for hours , days, weeks etc

### DISCUSSION THREAD

**Discussion by tarlen** [grafana/grafana#41666] [Dashboards, Panels & Visualizations]
From https://github.com/grafana/grafana/issues/15375

When using a time unit (eg minutes) in a panel, the display converts the time from a simple numerical value (eg 1440) into a much nicer human readable format (eg 1 day).

However the result does not correctly pluralise the output, leading to some rather awkward feeling results in the display.

<img width="115" alt="Screen Shot 2021-11-15 at 10 54 12 AM" src="https://user-images.githubusercontent.com/10524261/141700095-d664015b-ceda-43b6-94a9-5972774101f0.png">

In the above example all the of the values should be pluralised except the second to last (1 week).

Keep up the excellent work!

**Comment by leeoniya** [comment]
https://github.com/prantlf/fast-plural-rules
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/PluralRules/PluralRules

:thinking:

**Comment by ameinild** [comment]
I was wondering the exact same thing, but Tarlen beat me to it.

For some reason, it's working correctly with the "duration (s)" unit selected. So this should also be possible with the isolated time units (seconds, minutes) etc.

Also, the "rule" is quite easy, because when you have decimal time numbers, you (almost) never quite hit 1.00... (infinite). So actually, the time unit could always be plural.

0.8 hours
1.6 days
3.2 weeks
etc.

So someone just have to add the plural rule somewhere. 🤔

**Reply by ameinild** [reply]
The single logical exception could be when you choose to have 0 decimals - then you could argue that when the value is 1, plural should be omitted. But still, adding the plural form would be correct for the most part (and more correct than now, where it's wrong for the most part).

So the programming rule you want is: 
If decimals are zero AND if value is 1, then don't do anything - in any other case add plural form.

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
